File: 2357-make-array-zero-by-subtracting-equal-amounts/2357.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        h = []
        count = 0

        for n in nums:
            if n != 0:
                heappush(h, n)

        while len(h) > 0:
            x = heappop(h)
            h = []
            for i in range(len(nums)):
                if nums[i] > 0:
                    nums[i] -= x
                if nums[i] > 0:
                    heappush(h, nums[i])
            count += 1

        return count

File: 2357-make-array-zero-by-subtracting-equal-amounts/2357.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int count = 0;
        for (; count < nums.size(); ++count) {
            int minimum = INT_MAX;
            bool zeros = true;
            for (size_t i = 0; i < nums.size(); ++i) {
                if (nums[i] != 0 && nums[i] < minimum)
                    minimum = nums[i];
            }
            for (size_t i = 0; i < nums.size(); ++i) {
                if (nums[i] != 0) {
                    zeros = false;
                    nums[i] -= minimum;
                }
            }
            if (zeros)
                break;
        }
        return count;
    }
};
